ion-schema/src/system.rs

```diff
@@ -930,6 +938,34 @@ impl Resolver {
         }
         unresolvable_schema_error("Unable to load schema: ".to_owned() + id)
     }
+
+    /// Loads an [`IslSchema`] using authorities and type_store
+    // If we are loading the root schema then this will be set to `None` ( i.e. in the beginning when
+    // this method is called from the load_schema method of schema_system it is set to `None`)
+    // Otherwise if we are loading an import of the schema then this will be set to `Some(isl_import)`
+    // to be loaded (i.e. Inside schema_from_elements while loading imports this will be set to
+    // `Some(isl_import)`)
+    fn load_isl_schema<A: AsRef<str>>(
+        &mut self,
+        id: A,
+        load_isl_import: Option<&IslImport>,
+    ) -> IonSchemaResult<IslSchema> {
+        let id: &str = id.as_ref();
+
+        for authority in &self.authorities {
+            return match authority.elements(id) {
+                Err(error) => match error {
+                    IonSchemaError::IoError { source } => match source.kind() {
+                        ErrorKind::NotFound => continue,
+                        _ => Err(IonSchemaError::IoError { source }),
+                    },
+                    _ => Err(error),
+                },
+                Ok(schema_content) => self.isl_schema_from_elements(schema_content.into_iter(), id),
+            };
```

Comment:
Is it possible to do something like this so that we don't have so many nested `match`s?
```suggestion
            return match authority.elements(id) {
                Ok(schema_content) => self.isl_schema_from_elements(schema_content.into_iter(), id),
                Err(IonSchemaError::IoError { source: { kind: ErrorKind::NotFound, .. } }) => continue,
                err => err,
            };
```